main.test.js

```diff
@@ -0,0 +1,137 @@
+const request = require("supertest");
+const fsPromise = require("fs/promises");
+const app = require("./app");
+const { DB } = require("./api/shorturl");
+const expectResultGetHTMLStatistic = require("./utils");
+
+const existUrl = { url: "https://www.youtube.com/watch?v=tIuU_ra1YmY" };
+const newUrl = { url: "https://www.sport5.co.il/" };
+
+const expectedResultExistUrl = {
+  originalUrl: "https://www.youtube.com/watch?v=tIuU_ra1YmY",
+  shortUrlId: "1614871697604",
+  redirectCount: 0,
+  creationDate: "2021-3-4 17:28:17",
+};
+
+const invalidUrl = { url: "notValidUrl.com" };
+const messageInvalidUrl = { massage: `${new Error()} Invalid URL` };
+const messageNotFound = { message: `${new Error()} Short Url Is Not Found` };
+const messageNotFormat = { message: "ERROR: Wrong Format" };
+const messageNotExist = {
+  message: `${new Error()} Short Url Does Not Exist`,
+};
+
+// tests for Post Route Tests
+describe("Post Route Tests", () => {
+  it("Should return an error message with status code 400 for invalid url", async () => {
+    const response = await request(app)
+      .post("/api/shorturl/new")
+      .send(invalidUrl);
+
+    // Is the status code 400
+    expect(response.status).toBe(400);
+
+    // Is the body equal expectedResult
+    expect(response.body).toEqual(messageInvalidUrl);
+  });
+
+  it("Should get an existing url and return the same short url", async () => {
+    const response = await request(app)
+      .post("/api/shorturl/new")
+      .send(existUrl);
+    const response2 = await request(app)
+      .post("/api/shorturl/new")
+      .send(existUrl);
+    const existShortUrl = response2.body.shortUrlId;
+    const existCreationTime = response2.body.creationDate;
+
+    // Is the status code 200
+    expect(response2.status).toBe(200);
+
+    // Is the existShortUrl equal expectedResult shortUrlId
+    expect(existShortUrl).toEqual(expectedResultExistUrl.shortUrlId);
+
+    // Is the existCreationTime equal expectedResult creationDate
+    expect(existCreationTime).toEqual(expectedResultExistUrl.creationDate);
+  });
+
+  it("Should get a new url and add his object to the database", async () => {
+    let DBLengthBeforePost = DB.urlData.length;
+    const response = await request(app).post("/api/shorturl/new").send(newUrl);
+
+    // Is the length not equal after adding new Url
+    expect(DB.urlData.length).not.toBe(DBLengthBeforePost);
+  });
```

Comment:
You should also check a part of the response, just to be sure.
Also, don't use pascal case for variable names, only classes use this. (pascal case: `MyClass`, camel case: `myClass`)